.github/workflows/validate-protobufs.yml

```
@@ -0,0 +1,29 @@
+name: Validate Protobufs Compilation
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  validate-protobufs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get PR branch name
+        id: pr_branch
+        run: echo "branch=${GITHUB_HEAD_REF}" >> $GITHUB_ENV
+
+      - name: Checkout hedera-services
+        uses: actions/checkout@v3
+        with:
+          repository: hashgraph/hedera-services
+          path: hedera-services
+
+      - name: Update hedera-services branch
+        run: |
+          sed -i 's/branch = "main"/branch = "${{ env.branch }}"/' hedera-services/hapi/build.gradle.kts
```

This might have issues with the occasional branch modification.  Here's a possible fix:
```suggestion
          sed -i 's/branch = "[a-zA-Z\-\/0-9]+"/branch = "${{ env.branch }}"/' hedera-services/hapi/build.gradle.kts
```

---

Agreed. We might not have `main` as branch name always in services.